src/materialized/src/http.rs

```diff
@@ -140,30 +141,44 @@ impl Server {
         };
 
         let svc = service::service_fn(move |req| {
-            let user = match user {
-                Ok(ref user) => user,
-                Err(e) => {
-                    return async move { Ok(util::error_response(StatusCode::UNAUTHORIZED, e)) }
-                        .boxed()
+            let user = user.clone();
+            let coord_client = self.coord_client.clone();
+            let start_time = self.start_time;
+            async move {
+                let user = match user {
+                    Ok(user) => user,
+                    Err(e) => return Ok(util::error_response(StatusCode::UNAUTHORIZED, e)),
+                };
+
+                let coord_client = coord_client.new_conn()?;
+                let session = Session::new(coord_client.conn_id(), user);
+                let (coord_client, _) = match coord_client.startup(session).await {
+                    Ok(coord_client) => coord_client,
+                    Err(e) => {
+                        return Ok(util::error_response(
+                            StatusCode::INTERNAL_SERVER_ERROR,
+                            e.to_string(),
+                        ))
+                    }
+                };
+
+                match (req.method(), req.uri().path()) {
+                    (&Method::GET, "/") => root::handle_home(req, coord_client).await,
+                    (&Method::GET, "/metrics") => {
+                        metrics::handle_prometheus(req, coord_client, start_time).await
+                    }
+                    (&Method::GET, "/status") => {
+                        metrics::handle_status(req, coord_client, start_time).await
+                    }
+                    (&Method::GET, "/prof") => prof::handle_prof(req, coord_client).await,
+                    (&Method::GET, "/memory") => memory::handle_memory(req, coord_client).await,
+                    (&Method::POST, "/prof") => prof::handle_prof(req, coord_client).await,
+                    (&Method::POST, "/sql") => sql::handle_sql(req, coord_client).await,
+                    (&Method::GET, "/internal/catalog") => {
+                        catalog::handle_internal_catalog(req, coord_client).await
+                    }
+                    _ => root::handle_static(req, coord_client).await,
                 }
```

Comment:
Doesn't the session need to be terminated here?

Reply:
🙈